src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -150,7 +150,7 @@ private void initializeProducer() throws Exception {
   public void start() {
     if (_running.compareAndSet(false, true)) {
       initializeStateForPartitions();
-      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 40_000, 40_000, TimeUnit.MILLISECONDS);
+      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 1000, 30000, TimeUnit.MILLISECONDS);
```

Comment:
It might be worth waiting longer if there are many topics that need to be created concurrently.  In our internal configurations we start many monitors in the same process. And only one topic can be created a time because the controller does it.  Yes?